### PR TITLE
[native] Allow config values to have =

### PR DESCRIPTION
**Pull request by tanjialiang**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@tanjialiang Please, add a test to make sure this doesn't break in the future.

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp [hunk 1]
```diff
@@ -36,10 +35,10 @@ std::unordered_map<std::string, std::string> readConfig(
       continue;
     }
 
-    std::vector<std::string> configParts;
-    folly::split('=', line, configParts);
-    VELOX_USER_CHECK_EQ(configParts.size(), 2, "Malformed config: {}", line);
-    properties.emplace(configParts[0], configParts[1]);
+    auto delimiterPos = line.find('=');
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:38]: Should we check that delimiterPos != std::string::npos? Perhaps, add a test for invalid config.

File: presto-native-execution/presto_cpp/main/common/tests/ConfigTest.cpp [hunk 2]
```diff
@@ -191,4 +203,42 @@ TEST_F(ConfigTest, remoteFunctionServer) {
...
+      {"key", "value"},
+      {"emptyvaluekey", ""}};
+  ASSERT_EQ(configMap, expected);
+}
+
+TEST_F(ConfigTest, parseInvalid) {
+  auto testInvalid = [this](const std::string& fileContent) {
+    setUpConfigFilePath();
+    writeConfigFile(fileContent);
+    ASSERT_THROW(presto::util::readConfig(configFilePath), VeloxException);
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/common/tests/ConfigTest.cpp:238]: VELOX_ASSERT_THROW

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp [hunk 3]
```diff
@@ -36,10 +35,12 @@ std::unordered_map<std::string, std::string> readConfig(
...
+    VELOX_CHECK_NE(delimiterPos, std::string::npos);
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:39]: Let's add an error message with enough context to help troubleshooting.

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp [hunk 4]
```diff
@@ -36,10 +35,12 @@ std::unordered_map<std::string, std::string> readConfig(
...
+    auto name = line.substr(0, delimiterPos);
+    VELOX_CHECK(!name.empty());
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:41]: Should we trim the 'name'?

Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:41]: it's trimmed at the beginning of the loop

Comment (mbasmanova) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp]: Hmm... Let's say we have "   xxx = yyy  ". At the start of the loop we trim this to "xxx = yyy", then we get name="xxx " with space at the end, no?

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp [hunk 5]
```diff
@@ -36,10 +36,16 @@ std::unordered_map<std::string, std::string> readConfig(
       continue;
     }
 
-    std::vector<std::string> configParts;
-    folly::split('=', line, configParts);
-    VELOX_USER_CHECK_EQ(configParts.size(), 2, "Malformed config: {}", line);
-    properties.emplace(configParts[0], configParts[1]);
+    auto delimiterPos = line.find('=');
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:39]: nit: const auto delimiterPos =

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp [hunk 6]
```diff
@@ -36,10 +36,16 @@ std::unordered_map<std::string, std::string> readConfig(
...
+    VELOX_CHECK_NE(
+        delimiterPos,
+        std::string::npos,
+        "No '=' sign found for property pair '{}'",
+        line);
+    auto name = line.substr(0, delimiterPos);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/ConfigReader.cpp:45]: const nits.

**Review by xiaoxmeng — COMMENTED** [review]
@tanjialiang thanks!

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM. Thanks for the update!

File: presto-native-execution/presto_cpp/main/common/ConfigReader.cpp (hunk 4 above)
Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp]: at the start of the loop we get xxx=yyy so we get name = "xxx"

Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp]: The original logic essentially takes out all spaces in the line before proceeding to parse the key and values. I assume the assumption was made that no spaces are allowed in keys and values.

Comment (mbasmanova) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp]: Got it. Thank you for clarifying. It would be nice to document this behavior.

Comment (mbasmanova) [reply presto-native-execution/presto_cpp/main/common/ConfigReader.cpp]: Thinking about it again, this logic seems dangerous. Looks like it allows

```
foo=bar
f o o = b  ar
```

Does it make sense to allow spaces in config names? CC: @amitkdutta @xiaoxmeng

**Pull request merged**